app/test_main.py

```diff
@@ -2,0 +1,19 @@
+from unittest import mock
+
+from app.main import can_access_google_page
+
+
+@mock.patch("app.main.valid_google_url")
+def test_valid_google_url(
+        mocked_google_url: mock.MagicMock
+) -> None:
+    can_access_google_page("https://www.google.com/")
+    mocked_google_url.assert_called_once_with("https://www.google.com/")
+
+
+@mock.patch("app.main.has_internet_connection")
+def test_has_internet_connection(
```

Comment:
The test function name `test_has_internet_connection` is not descriptive enough. Consider renaming it to something like `test_can_access_google_page_with_internet_connection` to clearly indicate what the test is verifying.